README.md

```diff
@@ -1,72 +1,41 @@
+| --- | --- | --- | --- |
 
-### Install
 
-To support 'service principal' with certificate, ADAL depends on the 'cryptography' package. For smooth installation, some suggestions:
+The ADAL for Python library enables python applications to authenticate with Azure AD and get tokens to access Azure AD protected web resources.
 
-* For Windows and macOS
+You can learn in detail about ADAL Python functionality and usage documented in the [Wiki](https://github.com/AzureAD/azure-activedirectory-library-for-python/wiki).
 
-  Upgrade to the latest pip (8.1.2 as of June 2016) and just do `pip install adal`.
+## Versions
+Current version - 0.6.0
 
-* For Linux
+Minimum recommended version - 0.6.0
 
-  Upgrade to the latest pip (8.1.2 as of June 2016).
+You can find the changes for each version under [Releases](https://github.com/AzureAD/azure-activedirectory-library-for-python/releases).
 
-  You'll need a C compiler, libffi + its development headers, and openssl + its development headers.
-  Refer to [cryptography installation](https://cryptography.io/en/latest/installation/)
+> Note: Changes on 'client_id' and 'resource' arguments after 0.1.0
+>
+>The convenient methods in 0.1.0 have been removed, and now your application should provide parameter values to `client_id` and `resource`.
+>
+>2 Reasons:
+>
+>* Each adal client should have an Application ID representing a valid application registered in a tenant. The old methods borrowed the client-id of [azure-cli](https://github.com/Azure/azure-xplat-cli), which is never right. It is simple to register your application and get a client id. You can follow [this article](https://docs.microsoft.com/en-us/azure/active-directory/develop/active-directory-integrating-applications).
+>
+>* The old method defaults the `resource` argument to 'https://management.core.windows.net/', now you can just supply this value explictly. Please note, there are lots of different azure resources you can acquire tokens through adal though, for example, the samples in the repository acquire for the 'graph' resource. Because it is not an appropriate assumption to be made at the library level, we removed the old defaults.
 
-* To install from source:
-
-  Upgrade to the latest pip (8.1.2 as of June 2016).
-  To avoid dealing with compilation errors from cryptography, first run `pip install cryptography` to use statically-linked wheels.
-  Next, run `python setup.py install`
-
-  If you still like to build from source, refer to [cryptography installation](https://cryptography.io/en/latest/installation/).
-  For more context, start with this [stackoverflow thread](http://stackoverflow.com/questions/22073516/failed-to-install-python-cryptography-package-with-pip-and-setup-py).
-
-### Acquire Token with Client Credentials
```

Comment:
Although we do have and mention the "[sample applications on GitHub](https://github.com/azure-samples?utf8=%E2%9C%93&q=active-directory&type=&language=)" later in this README, but those samples locating in a separated repo may or may not cover all the scenarios ADAL Python supports (as of today, they don't). It would seem still beneficial to keep this minimal "out-of-box" samples sections here. What do you think?

Reply:
This content was moved to this  [Wiki page](https://github.com/AzureAD/azure-activedirectory-library-for-python/wiki/Acquire-tokens#adal-python-apis-for-corresponding-flows) which links to relavant dev sample for each flow. I will link to this page from Readme.

Reply:
The other 2 issues in this round of review are really minor, but this one is somewhat different, and needs more discussion.

Historically we had a set of real code snippet right here in the README, later we changed it into the current form as a link to those real samples, after we were called out that they became out-of-sync. See full details [here](https://github.com/AzureAD/azure-activedirectory-library-for-python/pull/103).

Now, moving these content to wiki page by actually duplicating those code snippets there, would bring us back to the potential 2-copies-would-become-out-of-sync situation. In fact, as a real-world example, one of our [current wiki page](https://github.com/AzureAD/azure-activedirectory-library-for-python/wiki/ADAL-basics#usage) technically already outdated, it includes an `api_version=None` which is no longer needed since our latest 1.0.0 release.

So my suggestion is, [anything that can go wrong will go wrong](https://en.wikipedia.org/wiki/Murphy%27s_law), so a better approach is to not create duplicate which we would forget to update later.